### PR TITLE
Update based on some recent developments

### PR DESCRIPTION
**Pull request by osorensen**

### CODE REVIEW

**Comment by osorensen** [comment]
@wleoncio the CodeFactor issues look like false positives to me, since the code it is referring to is in `work-docs`. Do you know how we can make CodeFactor ignore everything in that directory?

**Comment by wleoncio** [comment]
Sure, we could add something like this to [codecov.yml](https://github.com/ocbe-uio/BayesMallows/blob/master/codecov.yml)

```yaml
ignore:
    - "work-docs"
```

**Comment by wleoncio** [comment]
Strange. Maybe CF is not even reading the config file? 🤔

**Comment by wleoncio** [comment]
Just edited https://www.codefactor.io/repository/github/ocbe-uio/bayesmallows/ignore accordingly, may that'll do it.

bilde

**Comment by osorensen** [comment]
Could it be so that it checks the config file that is on the master branch??

**Comment by wleoncio** [comment]
> Could it be so that it checks the config file that is on the master branch??

I'm not even sure it's reading any file at all. On my other projects I don't even have a config file for CodeFactor  😅

**Comment by wleoncio** [comment]
Anyway, we can address all [code smells](https://www.codefactor.io/repository/github/ocbe-uio/bayesmallows/issues) on a different PR, I'll review the code now.

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+  sampling of latent ranks, specified in the "latent_sampling_lag" argument
+  to set_smc_options().
+* Prior for precision parameter alpha is now a gamma distribution. Until now
+  an exponential distribution has been assumed. Since the exponential is a special
+  case of the gamma with shape parameter equal to 1 (the default), this is not
+  a breaking change. However, it adds flexibility when it comes to specifying the prior.
+* setup_rank_data() now accepts a single vector of rankings, silently converting a to matrix with a single row.
+* Sequential Monte Carlo algorithm can now start from a sample from the prior
+  distribution, see the sample_prior() function for an example.
+* Added support for parallelism under-the-hood with oneTBB.
```
Comment (wleoncio) [inline NEWS.md:16]: Missing space on "oneTBB"? Might be an excuse to trigger another CodeFactor run 😆

Comment (osorensen) [reply NEWS.md:16]: Unfortunately not ;-)

https://github.com/oneapi-src/oneTBB

Comment (osorensen) [reply NEWS.md:16]: But the fix to #381 seems pretty easy, so I'll try to add that one to this PR and trigger a new run.

File: tests/testthat/test-compute_posterior_intervals.R [hunk 2]
```diff
@@ -71,17 +71,17 @@ test_that("compute_posterior_intervals works for SMC", {
...
   mod3 <- update_mallows(
     mod2,
     new_data = setup_rank_data(potato_visual[10:12, ])
   )
 
   pi <- compute_posterior_intervals(mod3, decimals = 2)
-  expect_equal(pi$hpdi, "[2.18,3.75]")
+  expect_equal(pi$hpdi, "[2.23,3.74]")
   pi <- compute_posterior_intervals(mod3, parameter = "rho")
   expect_equal(pi$hpdi[[20]], "[1,5]")
```
Comment (wleoncio) [inline tests/testthat/test-compute_posterior_intervals.R:87]: This is alright, but for future-proofing, perhaps we could compare numbers with a high tolerance (like 0.1)?

**Review by wleoncio — APPROVED** [review]
Good job!

File: tests/testthat/test-compute_posterior_intervals.R (hunk 2 above)
Comment (osorensen) [reply tests/testthat/test-compute_posterior_intervals.R:87]: I agree. I just have to figure out the random number generation stuff on my M1 mac. On all other platforms the results are reproduced, but M1 mac seems to work differently, even though I use Rcpp's random number generators which are designed to by platform independent. I don't know if it's the RNGs fault or if there are differences in floating point representations or something like that which causes the input to the functions for generating random numbers to differ. I think my next laptop will run Linux.

**Comment by osorensen** [comment]
Thanks @wleoncio! I added #382 which is a simple fix.

**Comment by wleoncio** [comment]
No problem, looks good. We'll fight CodeFactor later...

**Pull request merged**